### PR TITLE
fix(openai): allow tokens or token arrays as input for Embeddings endpoint

### LINKED ISSUE #5884: "trunc()" function (contrib/openai /patch.py) is expecting a string instead of a list
### Summary of problem

Instrumentation of OpenAI by ddtrace is failing when invoking the "trunc()" function (contrib/openai /patch.py)). ddtrace is expecting a string while a list is returned. The issue seems to appear at that [line](https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/contrib/openai/patch.py#L542).

### Which version of dd-trace-py are you using?

1.13.3

### Which version of pip are you using?

23.1.2

### Which libraries and their versions are you using?

<details>
  <summary>`pip freeze`</summary>
  aiohttp==3.8.4
aiosignal==1.3.1
anyio==3.6.2
async-timeout==4.0.2
attrs==23.1.0
beautifulsoup4==4.12.2
bs4==0.0.1
bytecode==0.14.1
cattrs==22.2.0
certifi==2023.5.7
charset-normalizer==3.1.0
click==8.1.3
dataclasses-json==0.5.7
ddsketch==2.0.4
ddtrace==1.12.0
Deprecated==1.2.13
dnspython==2.3.0
envier==0.4.0
exceptiongroup==1.1.1
fastapi==0.95.0
frozenlist==1.3.3
greenlet==2.0.2
h11==0.14.0
idna==3.4
importlib-metadata==6.0.1
jsonschema==4.17.3
langchain==0.0.136
loguru==0.7.0
marshmallow==3.19.0
marshmallow-enum==1.5.1
multidict==6.0.4
mypy-extensions==1.0.0
numpy==1.24.3
openai==0.27.4
openapi-schema-pydantic==1.2.4
opentelemetry-api==1.17.0
packaging==23.1
pandas==1.5.3
pinecone-client==2.2.1
protobuf==4.23.0
pydantic==1.10.7
pyrsistent==0.19.3
python-dateutil==2.8.2
python-dotenv==1.0.0
pytz==2023.3
PyYAML==6.0
regex==2023.5.5
requests==2.30.0
six==1.16.0
slack-bolt==1.17.1
slack-sdk==3.21.3
sniffio==1.3.0
soupsieve==2.4.1
SQLAlchemy==1.4.48
starlette==0.26.1
tenacity==8.2.2
tiktoken==0.3.3
tqdm==4.65.0
typing-inspect==0.8.0
typing_extensions==4.5.0
urllib3==2.0.2
uvicorn==0.21.1
wrapt==1.15.0
xmltodict==0.13.0
yarl==1.9.2
zipp==3.15.0
</details>

### How can we reproduce your problem?

```
from langchain import PromptTemplate
from langchain.chains import RetrievalQAWithSourcesChain
from langchain.chains.qa_with_sources.base import BaseQAWithSourcesChain
from langchain.chat_models import ChatOpenAI
from vectors import retriever
from ddtrace import tracer

def create():
    llm = ChatOpenAI(model_name=model)  # type: ignore
    template = """Test"""
    prompt = PromptTemplate(template=template, input_variables=["summaries", "question"])

    chain_type_kwargs = {"prompt": prompt}

    return RetrievalQAWithSourcesChain.from_chain_type(
        llm=llm, chain_type="stuff", retriever=retriever(index_name=index_name, language=language), chain_type_kwargs=chain_type_kwargs
    )
    
qa = create(index_name, language=language, model="gpt-4")
with tracer.trace("question_answering"):
  result = qa({"question": question}, return_only_outputs=True)
```

### What is the result that you get?

```


"AttributeError: 'list' object has no attribute 'replace'"
--
"text = text.replace(\"\n\", \"\\\\n\").replace(\"\\t\", \"\\\\t\")"
"File \"/usr/local/lib/python3.9/site-packages/ddtrace/contrib/openai/patch.py\", line 207, in trunc"
"span.set_tag_str(\"openai.request.input.%d\" % idx, integration.trunc(inp))"
"File \"/usr/local/lib/python3.9/site-packages/ddtrace/contrib/openai/patch.py\", line 633, in handle_request"
"hook.send(None)"
"File \"/usr/local/lib/python3.9/site-packages/ddtrace/contrib/openai/patch.py\", line 306, in _traced_endpoint"
"g.send(None)"
"File \"/usr/local/lib/python3.9/site-packages/ddtrace/contrib/openai/patch.py\", line 335, in patched_endpoint"
"@classmethod"
"File \"/usr/local/lib/python3.9/site-packages/openai/api_resources/embedding.py\", line 14, in create"
"return embeddings.client.create(**kwargs)"
"File \"/usr/local/lib/python3.9/site-packages/langchain/embeddings/openai.py\", line 51, in _completion_with_retry"
"result = fn(*args, **kwargs)"
"File \"/usr/local/lib/python3.9/site-packages/tenacity/__init__.py\", line 382, in __call__"
"raise self._exception"
"File \"/usr/local/lib/python3.9/concurrent/futures/_base.py\", line 391, in __get_result"
"return self.__get_result()"
"File \"/usr/local/lib/python3.9/concurrent/futures/_base.py\", line 439, in result"
"return fut.result()"
"File \"/usr/local/lib/python3.9/site-packages/tenacity/__init__.py\", line 314, in iter"
"do = self.iter(retry_state=retry_state)"
"File \"/usr/local/lib/python3.9/site-packages/tenacity/__init__.py\", line 379, in __call__"
"return self(f, *args, **kw)"
"File \"/usr/local/lib/python3.9/site-packages/tenacity/__init__.py\", line 289, in wrapped_f"
"return _completion_with_retry(**kwargs)"
"File \"/usr/local/lib/python3.9/site-packages/langchain/embeddings/openai.py\", line 53, in embed_with_retry"
"response = embed_with_retry("
"File \"/usr/local/lib/python3.9/site-packages/langchain/embeddings/openai.py\", line 206, in _get_len_safe_embeddings"
"return self._get_len_safe_embeddings([text], engine=engine)[0]"
"File \"/usr/local/lib/python3.9/site-packages/langchain/embeddings/openai.py\", line 236, in _embedding_func"
"embedding = self._embedding_func(text, engine=self.query_model_name)"
"File \"/usr/local/lib/python3.9/site-packages/langchain/embeddings/openai.py\", line 280, in embed_query"
"query_obj = self._embedding_function(query)"
"File \"/usr/local/lib/python3.9/site-packages/langchain/vectorstores/pinecone.py\", line 148, in similarity_search"
"docs = self.vectorstore.similarity_search(query, **self.search_kwargs)"
"File \"/usr/local/lib/python3.9/site-packages/langchain/vectorstores/base.py\", line 225, in get_relevant_documents"
"docs = self.retriever.get_relevant_documents(question)"
"File \"/usr/local/lib/python3.9/site-packages/langchain/chains/qa_with_sources/retrieval.py\", line 45, in _get_docs"
"docs = self._get_docs(inputs)"
"File \"/usr/local/lib/python3.9/site-packages/langchain/chains/qa_with_sources/base.py\", line 118, in _call"
"outputs = self._call(inputs)"
"File \"/usr/local/lib/python3.9/site-packages/langchain/chains/base.py\", line 113, in __call__"
"raise e"
"File \"/usr/local/lib/python3.9/site-packages/langchain/chains/base.py\", line 116, in __call__"
"result = qa({\"question\": question}, return_only_outputs=True)"
"File \"/code/slack.py\", line 173, in answer_question"
"result = await answer_question(question, language)"
"File \"/code/slack.py\", line 98, in handle_mention"
"return await self.ack_function("
"File \"/usr/local/lib/python3.9/site-packages/slack_bolt/listener/async_listener.py\", line 119, in run_ack_function"
"await listener.run_ack_function(request=request, response=response)"
"File \"/usr/local/lib/python3.9/site-packages/slack_bolt/listener/asyncio_runner.py\", line 118, in run_ack_function_asynchronously"
"Traceback (most recent call last):"
"Failed to run listener function (error: 'list' object has no attribute 'replace')"
```

Error at [line](https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/contrib/openai/patch.py#L542) 
image

### What is the result that you expected?

No error

### PR DESCRIPTION
**Pull request by Yun-Kim**
Fixes #5884.

This PR adds handling for non-string request input params for the `Embeddings.create()` method. This API method accepts strings, array of strings, as well as token arrays (list of ints) and arrays of token arrays (list of list of ints), but the previous implementation only accounted for strings and array of strings. Token arrays or arrays of token arrays being passed in as arguments would then cause errors as the traced request handler for the Embeddings endpoint did not account for non-string types.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] OPTIONAL: PR description includes explicit acknowledgement of the performance implications of the change as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

**Review by emmettbutler — APPROVED** [review]

File: releasenotes/notes/fix-openai-embeddings-input-7feb1198fabac31b.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    openai: Resolved an issue where using an array of tokens or an array of token arrays as the
```
Comment (mabdinur) [inline releasenotes/notes/fix-openai-embeddings-input-7feb1198fabac31b.yaml:4]: nit: tense, also can we simplify the release note a bit. It's a bit wordy 
```suggestion
    openai: Resolves an issue where using an array of tokens or an array of token arrays as the 
```

**Review by mabdinur — APPROVED** [review]
